src/main/java/build/buildfarm/instance/shard/ShardInstance.java

```diff
@@ -762,6 +768,7 @@ public void onNext(ByteString nextChunk) {
                 blobObserver.onNext(nextChunk);
                 received += nextChunk.size();
                 ioMetric.observe(received);
+                casHitCounter.inc();
```

Comment:
This will need to go in the onCompleted() below, otherwise we'll be incrementing on every chunk received.

Reply:
Thanks, I thought that we want the successful chunks.